xapian-core/include/xapian/types.h

```diff
@@ -30,6 +30,13 @@
 
 namespace Xapian {
 
+/**A unique identifier for a cluster of documents
+ *
+ * clusterid of 0 is invalid, providing an "out of range" value which can
+   be used to mean "not a valid cluster"
+ */
+typedef unsigned XAPIAN_DOCID_BASE_TYPE clusterid;
```

Comment:
`XAPIAN_DOCID_BASE_TYPE` isn't appropriate for this. Do you need a signed type? It's probably best to use `int` or `unsigned int`.